### PR TITLE
Add man page for monerod !

### PR DESCRIPTION
**Pull request by Justxd22**
All the essential `monerod` documentation, one quick place for docs without leaving your terminal:

- Includes `Options` and `daemon_commands`
- Comprehensive list of all `JSON RPC` methods
- Detailed `Example` section
- Code blocks included for clarity

This allows us to efficiently direct someone to a specific section of the man page, rather than vaguely referring them to "somewhere in the middle of the --help subcommand."

The man page will be bundled with the tarball releases, making it as simple as running man `./docs/man/monerod.1`.

### CODE REVIEW

**Comment by iamamyth** [comment]
Thanks for putting this together; it looks quite thorough and really does improve the user experience.

**Comment by Justxd22** [comment]
thanks :D

**Comment by selsta** [comment]
Could you squash the commits together?

**Comment by Justxd22** [comment]
@selsta done

File: docs/man/monerod.1 [hunk 1]
```diff
@@ -0,0 +1,5343 @@
+.TH MONEROD "1" "AUGUST 2024" "Fluorine Fermi 0.18.3.3" "MONEROD MANUAL"
+.SH NAME
+monerod \- Connects you to Monero network
+.SH SYNOPSIS
+.B monerod
+[\fI\,options|settings\/\fR] [\fI\,daemon_command...\/\fR]
+.SH DESCRIPTION
+.\" Add any additional description here
+.PP
+The Monero daemon \fBmonerod\fR keeps your computer synced up with the Monero network.
```
Comment (iamamyth) [inline docs/man/monerod.1:10]: Suggest using the full word (`synchronized`) here and removing "up": `keeps your computer synchronized with`

File: docs/man/monerod.1 [hunk 2]
```diff
@@ -0,0 +1,5343 @@
...
+
+It downloads and validates the blockchain from the p2p network.
+
+\fBmonerod\fR is entirely decoupled from your wallet.
+
+\fBmonerod\fR is mainly used by your wallet software over local RPC connection, as in \fBMonero GUI\fR.
+
+\fBmonerod\fR does not access your private keys \- it is not aware of your transactions and balance.
+
+This allows you to run \fBmonerod\fR on a separate computer or in the cloud.
```
Comment (iamamyth) [inline docs/man/monerod.1:20]: This sentence should be part of the preceding paragraph, otherwise you have "this" without any antecedent. Relatedly, I would suggest rewording, for clarity:
```
\fBmonerod\fR does not access your private keys \- it is not aware of your transactions and balance, so you can run \fBmonerod\fR on a separate computer or in the cloud.
```

You can then also join this reworded intro with the following paragraph (note some minor grammar edits):
```
\fBmonerod\fR does not access your private keys \- it is not aware of your transactions and balance, so you can run \fBmonerod\fR on a separate computer or in the cloud. In fact, you can connect to a remote \fBmonerod\fR instance provided by a semi-trusted 3rd party, who will not be able to steal your funds; this third-party configuration is very handy for learning and experimentation.
```

File: docs/man/monerod.1 [hunk 3]
```diff
@@ -0,0 +1,5343 @@
...
+
+In fact, you can connect to a remote \fBmonerod\fR instance provided by a semi-trusted 3rd party. Such 3rd party will not be able to steal your funds. This is very handy for learning and experimentation.
+
+However, there are privacy and reliability implications to using a remote, untrusted node. For any real business you should be running your own full node.
```
Comment (iamamyth) [inline docs/man/monerod.1:24]: Would probably make this sentence the conclusion of the prior paragraph on remote/3rd party nodes.

File: docs/man/monerod.1 [hunk 4]
```diff
@@ -0,0 +1,5343 @@
...
+
+\fBmonerod\fR can save you time syncing the whole blockchain by relaying your wallet requests over p2p network nodes 
+.br
+by using \fB--bootstrap-daemon-address\fR see \fBmonerod Performance\fR section for more info.
+
+
+.SH Options:
+Arguments in brackets are default
```
Comment (iamamyth) [inline docs/man/monerod.1:32]: I think brackets here should say parentheses.

Comment (iamamyth) [reply docs/man/monerod.1:32]: To clarify: `[` is a backet, `(` is a parentesis, `(` is sometimes called a "round bracket" but the term "bracket" without any adjective normally means a square bracket (and is too ambiguous because the documentation contains both round and square brackets).

File: docs/man/monerod.1 [hunk 5]
```diff
@@ -0,0 +1,5343 @@
+.TH MONEROD "1" "AUGUST 2024" "Fluorine Fermi 0.18.3.3" "MONEROD MANUAL"
+.SH NAME
+monerod \- Connects you to Monero network
```
Comment (iamamyth) [inline docs/man/monerod.1:3]: Daemon descriptions in man pages tend to be a bit more precise, I might suggest:
```
Monero p2p network daemon
```

File: docs/man/monerod.1 [hunk 6]
```diff
@@ -0,0 +1,5343 @@
...
+\fB\--detach\fR
+Run as daemon
+.TP
+\fB\--non-interactive\fR                     
+Run non-interactive, Do not require tty in a foreground mode. Helpful when running in a container. By default \fBmonerod\fR runs in a foreground and opens stdin for reading.
+.TP
+\fB\--no-zmq\fR
+Disable ZMQ RPC server
+.TP
+\fB\--no-igd\fR
```
Comment (iamamyth) [inline docs/man/monerod.1:93]: This should probably be in the same section as `--igd`

File: docs/man/monerod.1 [hunk 7]
```diff
@@ -0,0 +1,5343 @@
...
+\fB\--fast-block-sync\fR arg (=1)
+Sync up most of the way by using embedded, known block hashes.
+.TP
+\fB\--block-sync-size\fR arg (=0)
+How many blocks to sync at once during chain synchronization (0 = adaptive).
+.TP
+\fB\--block-download-max-size\fR arg (=0)
+Set maximum size of block download queue in bytes (0 for default)
+.TP
+\fB\--no-sync\fR
```
Comment (iamamyth) [inline docs/man/monerod.1:156]: These options (especially the ones starting with extra-mesages-file) aren't really performance related.  I suggest folding them into the core server config section.

File: docs/man/monerod.1 [hunk 8]
```diff
@@ -0,0 +1,5343 @@
...
+\fB\--regtest\fR
+Run in a regression testing mode.
+.TP
+\fB\--fixed-difficulty\fR arg (=0)
+Fixed difficulty used for testing.
+
+
+.SS Legacy Options
+.TP
+\fB\--ban-list\fR arg
```
Comment (iamamyth) [inline docs/man/monerod.1:419]: With the exception of fluffy-blocks, these are not "legacy options", and some, such as ban-list, are currently suggested best practice to defend against network snooping by fleets node proxies. I would suggest just folding all of these into the server config section, though if you want to keep [no-]fluffy-blocks here, that's fine.

File: docs/man/monerod.1 [hunk 9]
```diff
@@ -0,0 +1,5343 @@
...
+Get or set the download limit.
+.TP
+\fBlimit_up [<kB/s>]\fR
+Get or set the upload limit.
+
+
+
+.SS Transaction pool:
+.TP
+\fBflush_cache [bad-txs] [bad-blocks]\fR
```
Comment (iamamyth) [inline docs/man/monerod.1:509]: I wouldn't consider `flush_cache bad-blocks` a transaction pool command; would suggest moving `flush_cache` to the blockchain section.

File: docs/man/monerod.1 [hunk 10]
```diff
@@ -0,0 +1,5343 @@
...
+
+Mind blockchain reorganizations. Block notifications can revert to same and past heights. Small reorganizations are natural and happen every day.
+
+
+.SS Change blocks sync size:
+How many blocks are processed in a single batch during chain synchronization. By default this is 20 blocks for newer history and 100 blocks for older history (“pre v4”). Default behavior is represented by value 0. Intuitively, the more resources you have, the bigger batch size you may want to try out.
+
+.B ./monerod --block-sync-size=500
+
+.SS bootstrap:
```
Comment (iamamyth) [inline docs/man/monerod.1:5280]: Should probably capitalize the title here (`Bootstrap`).

File: docs/man/monerod.1 [hunk 11]
```diff
@@ -0,0 +1,5343 @@
...
+mainnet is when you want to deal with the real XMR.
+
+\fB\./monerod \--detach\fR        # run as a daemon in background
+.br
+\fBtail \-f ~/.bitmonero/bitmonero.log\fR  # watch the logs
+.br
+\fB\./monerod exit\fR           # ask daemon to exit gracefully
+
+
+.SS p2p network protocol over Tor via torsocks:
```
Comment (iamamyth) [inline docs/man/monerod.1:5228]: Most conventions would consider `P2P` an acronym and capitalize it throughout the document; I don't mind keeping it lower case, but, as the first word in a section heading, it should have the first letter capitalized (`P2p`)

Comment (iamamyth) [reply docs/man/monerod.1:5228]: Note: The prior comment applies to the following section, as well.

**Comment by iamamyth** [comment]
There's a case to be made that the RPC documentation should be a separate man page which the main man page references; I don't have a strong opinion on the matter, though I will note that the RPC tends to change more regularly than the core daemon, which might slightly strengthen the case for a split.

File: docs/man/monerod.1 (hunk 8 above)
Comment (nahuhh) [reply docs/man/monerod.1:419]: Probably pulled / transposed some info from old monerodocs.org 

docs.getmonero.org updated this info to properly categorize some of the so-called "legacy" options

**Comment by Justxd22** [comment]
@iamamyth about the separate RPC man page, i don't like the idea of having 2 man pages and having to switch between them, or if someone refers you to x section in the man page they have to specify which one of the two, BUT if it's needed happy to do it

File: docs/man/monerod.1 (hunk 8 above)
Comment (Justxd22) [reply docs/man/monerod.1]: i only used getmonero.dev and getmonero.org for info, and they still list them in legacy looks like they need updating as well

Comment (nahuhh) [reply docs/man/monerod.1]: Docs.getmonero.org does not list as legacy

Comment (Justxd22) [reply docs/man/monerod.1]: yup updated them

File: docs/man/monerod.1 [hunk 12]
```diff
@@ -0,0 +1,5338 @@
...
+.TP
+\fB\--enforce-dns-checkpointing\fR
+checkpoints from DNS server will be enforced, The emergency checkpoints set by \fBMoneroPulse\fR operators will be enforced. It is probably a good idea to set enforcing for unattended nodes.
+If encountered block hash does not match corresponding checkpoint, the local blockchain will be rolled back a few blocks, effectively blocking following what MoneroPulse operators consider invalid fork. The log entry will be produced: \fBERROR Local blockchain failed to pass a checkpoint, rolling back!\fR Eventually, the alternative (“fixed”) fork will get heavier and the node will follow it, leaving the “invalid” fork behind.
+.TP
+\fB\--disable-dns-checkpoints\fR
+Do not retrieve checkpoints from DNS
+.TP
+\fB\--ban-list\fR arg
+Specify ban list file, one IP address per line
```
Comment (iamamyth) [inline docs/man/monerod.1:97]: Per comment by `nahuhh`, apparently the other monero docs list `ban-list` under p2p rather than core server options (this would also apply to `enable-dns-blocklist`). I can see a case for either choice (if memory serves, the bans only apply to p2p, if they also applied to RPC the options would definitely belong here).

File: docs/man/monerod.1 [hunk 13]
```diff
@@ -0,0 +1,5338 @@
...
+monerod \- Monero p2p network daemon
+.SH SYNOPSIS
+.B monerod
+[\fI\,options|settings\/\fR] [\fI\,daemon_command...\/\fR]
+.SH DESCRIPTION
+.\" Add any additional description here
+.PP
+The Monero daemon \fBmonerod\fR keeps your computer synchronized with the Monero network.
+
+It downloads and validates the blockchain from the p2p network.
```
Comment (iamamyth) [inline docs/man/monerod.1:12]: Would fold into preceding sentence:
```
The Monero daemon \fBmonerod\fR keeps your computer synchronized with the Monero blockchain by downloading and validating the blockchain from the p2p network.
```

**Comment by iamamyth** [comment]
A couple of (I suspect final) minor comments, but this looks good.

File: docs/man/monerod.1 (hunk 12 above)
Comment (nahuhh) [reply docs/man/monerod.1:97]: I just tested and it looks like it blocks rpc as well

Comment (Justxd22) [reply docs/man/monerod.1:97]: so @nahuhh i should keep it as is ?

Comment (Justxd22) [reply docs/man/monerod.1:97]: @iamamyth what about this? as per @nahuhh it's p2p & rpc, so i should keep it as is

Comment (iamamyth) [reply docs/man/monerod.1:97]: Yes keep as-is.

**Review by iamamyth — APPROVED** [review]
Please squash commits.

File: docs/man/monerod.1 (hunk 10 above)
Comment (nahuhh) [reply docs/man/monerod.1]: block-sync-size=500 wont work btw. Max is 100 due to some broken code. If you edit code to allow >100, you'll only be able to sync >100 from node who have also run the updated version

Comment (Justxd22) [reply docs/man/monerod.1]: @nahuhh Does the official docs say so?

Comment (nahuhh) [reply docs/man/monerod.1]: Does it say that the max is 100? No. Because its a bug. The max is supposed to be 2048 but it wont ever switch to that unless done so in a hard fork.

You can probably leave the example as 500, assuming it will be fixed in the upcoming HF (it hasnt been yet)

https://github.com/nahuhh/monero/commit/52d9682aba133779bf53b37318462d1b0d8f9216

**Comment by nahuhh** [comment]
@Justxd22 there have been some changes and additions, but i guess can be done in a followup pr

```
\fB\--limit-rate-up\fR arg (=2048)
 set limit-rate-up [kB/s]
.TP
\fB\--limit-rate-down\fR arg (=8192)
 set limit-rate-down [kB/s]
```

is now 8192 and 32768

there are new flags like --batch-max-weight (related to block-sync-size) and --span-limit (related to block-download-max-size). (neither of these is merged on docs [yet](https://github.com/monero-project/monero-docs/pull/271))

I'm sure im missing some other changes as well, but those are the 2 that immediately come to mind

**Comment by Justxd22** [comment]
@nahuhh  should i add them now? or a follow-up PR?

**Comment by selsta** [comment]
please add it to this PR

**Comment by nahuhh** [comment]
@Justxd22 drop the merge commit

`git reset --hard HEAD~1`

I dont think this PR will need a rebase, so no need to update the base

**Comment by Justxd22** [comment]
@nahuhh i added couple of missing flags as well, it doesn't show the changes it was squashed...

```diff

diff --git a/docs/man/monerod.1 b/docs/man/monerod.1
index 426d45a37..f18cf2e3c 100644
--- a/docs/man/monerod.1
+++ b/docs/man/monerod.1
@@ -109,6 +109,9 @@ Port for ZMQ RPC server to listen on
 \fB\--block-download-max-size\fR arg (=0)
 Set maximum size of block download queue in bytes (0 for default)
 .TP
+\fB\--span-limit\fR arg (=2)
+Dynamically restrict the number of spans in queue by estimating how many spans can be synced within N minutes. The dynamic minimum is 10 spans, and is further capped by block-download-max-size.
+.TP
 \fB\--no-sync\fR                   
 Don't synchronize the blockchain with other peers, intended to be used with \fB\--bootstrap-daemon-address\fR allowing reliance on other public P2P nodes for blockchain data.
 
@@ -160,7 +163,10 @@ Max number of threads to use when preparing block hashes in groups.
 Sync up most of the way by using embedded, known block hashes.
 .TP
 \fB\--block-sync-size\fR arg (=0)
-How many blocks to sync at once during chain synchronization (0 = adaptive).
+How many blocks are processed in a single batch during chain synchronization. Setting this flag will override the default behavior, which is to be calculated dynamically by batch-max-weight. Max value is 100.
+.TP
+\fB\--batch-max-weight\fR arg (=10)
+Dynamically set the block-sync-size by targetting this many MB. Max value is 50.
 .TP
 \fB\--bootstrap-daemon-address\fR arg
 URL of a 'bootstrap' remote daemon that the connected wallets can use while this daemon is still not fully synced. Use 'auto' to enable automatic public nodes discovering and bootstrap daemon switching,
@@ -241,10 +247,10 @@ Specify list of peers to connect to only. If this option is given the options ad
 \fB\--in-peers\fR arg (=-1)
  set max number of in peers
 .TP
-\fB\--limit-rate-up\fR arg (=2048)
+\fB\--limit-rate-up\fR arg (=8192)
  set limit-rate-up [kB/s]
 .TP
-\fB\--limit-rate-down\fR arg (=8192)
+\fB\--limit-rate-down\fR arg (=32768)
  set limit-rate-down [kB/s]
 .TP
 \fB\--limit-rate\fR arg (=-1)
@@ -301,6 +307,18 @@ Ignore unsuccessful IPv4 bind for RPC
 \fB\--restricted-rpc\fR
 Restrict RPC to view only commands and do not return privacy sensitive data in RPC calls
 .TP
+\fB\--rpc-max-connections\fR arg (=100)
+Max RPC connections permitted
+.TP
+\fB\--rpc-max-connections-per-public-ip\fR arg (=3)
+Max RPC connections per public IP permitted
+.TP
+\fB\--rpc-max-connections-per-private-ip\fR arg (=25)
+Max RPC connections per private and localhost IP permitted
+.TP
+\fB\--rpc-response-soft-limit\fR arg (=26214400)
+Max response bytes that can be queued, enforced at next response attempt
+.TP
 \fB\--rpc-restricted-bind-ip\fR arg (=127.0.0.1)
 Specify IP to bind restricted RPC server
 .TP

```

**Comment by nahuhh** [comment]
https://github.com/monero-project/monero/pull/10388
